streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/cluster/ServiceBundle.java

```diff
@@ -0,0 +1,210 @@
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+ **/
+package com.hortonworks.streamline.streams.catalog.cluster;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hortonworks.streamline.common.Schema;
+import com.hortonworks.streamline.storage.PrimaryKey;
+import com.hortonworks.streamline.storage.Storable;
+import com.hortonworks.streamline.storage.StorableKey;
+import com.hortonworks.streamline.streams.catalog.topology.TopologyComponentUISpecification;
+import com.hortonworks.streamline.streams.layout.storm.FluxComponent;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ServiceBundle implements Storable {
+
+    public static final String NAME_SPACE = "service_bundle";
+    public static final String ID = "id";
+    public static final String NAME = "name";
+    public static final String REGISTER_CLASS = "registerClass";
+    public static final String UI_SPECIFICATION = "serviceUISpecification";
+    public static final String TIMESTAMP = "timestamp";
+
+    /**
+     * Unique id for a service bundle. This is the primary key
+     */
+    private Long id;
+
+    /**
+     * The service name. For e.g. STORM, KAFKA, HDFS, etc.
+     */
+    private String name;
+
+    /**
+     * Time recording the creation or last update of this instance
+     */
+    private Long timestamp;
+
+    /**
+     * Object that will be used by ui to elicit values from user for registering this service
+     */
+    private TopologyComponentUISpecification serviceUISpecification;
+
+    /**
+     * A fully qualified class name that can help registering the service.
+     */
+    private String registerClass;
+
+    @Override
+    @JsonIgnore
+    public String getNameSpace () {
+        return NAME_SPACE;
+    }
+
+    @Override
+    @JsonIgnore
+    public Schema getSchema () {
+        return Schema.of(
+                new Schema.Field(ID, Schema.Type.LONG),
+                new Schema.Field(NAME, Schema.Type.STRING),
+                new Schema.Field(TIMESTAMP, Schema.Type.LONG),
+                new Schema.Field(UI_SPECIFICATION, Schema.Type.STRING),
+                Schema.Field.optional(REGISTER_CLASS, Schema.Type.STRING)
+        );
+    }
+
+    @Override
+    @JsonIgnore
+    public PrimaryKey getPrimaryKey () {
+        Map<Schema.Field, Object> fieldToObjectMap = new HashMap<Schema.Field, Object>();
+        fieldToObjectMap.put(new Schema.Field(ID, Schema.Type.LONG),
+                this.id);
+        return new PrimaryKey(fieldToObjectMap);
+    }
+
+    @Override
+    @JsonIgnore
+    public StorableKey getStorableKey () {
+        return new StorableKey(getNameSpace(), getPrimaryKey());
+    }
+
+    @Override
+    public Map toMap () {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Agreed on static field. It shouldn't be an instance field since it would be on every entities.